cardano-cli/src/Cardano/CLI/Shelley/Run/Genesis.hs

```diff
@@ -1376,6 +1375,7 @@ readProtocolParametersSourceSpec (ParamsFromFile f) = readProtocolParameters f
 
 --TODO: eliminate this and get only the necessary params, and get them in a more
 -- helpful way rather than requiring them as a local file.
+{-# DEPRECATED readProtocolParameters "Query the node instead of using a parameters file" #-}
```

Comment:
Sorry, I wasn't clear. It's fine to read the `ProtocolParameters` from file. What it's not longer ok to do is to derive the `ProtocolParameters` from a `ShelleyGensis` because there have been changes to the `ProtocolParameters` since the `ShelleyEra` (specifically in Alonzo and Babbage eras). So this deprecation notice belongs by `readProtocolParametersSourceSpec` and we should replace all calls of `readProtocolParametersSourceSpec` with `readProtocolParameters`.

Reply:
I see. I think `readProtocolParametersSourceSpec` is a good place to have strategy pattern for the source of `ProtocolParams`.  - having in mind the next step, in which we should provide a new option for querying node state. Maybe we could leave it as is, remove `ParamsFromGenesis`   and corresponding CLI option?
@Jimbo4350 would that work, or would you suggest implementing source selection in a different place?

Reply:
>  I think readProtocolParametersSourceSpec is a good place to have strategy pattern for the source of ProtocolParams.

I don't understand what you mean by this.

> remove ParamsFromGenesis and corresponding CLI option 

This is what we should do. Replace all calls to `readProtocolParametersSourceSpec` with `readProtocolParameters` and adjust the cli parsers accordingly. It's ok to break this because using `readProtocolParametersSourceSpec` with a shelley genesis file will create an incorrect `ProtocolParameters` value. Once that is done and merged, we can talk about using a node state query.

Reply:
@Jimbo4350 what about ` ProtocolParamsSourceSpec` ? It will become unused after that change, should I get rid of it as well?

Reply:
Yep, get rid of it.